[12.0][MIG] backport to 12.0

huguesdk opened the pull request:
*   backport to 12.0.
*   add new module `account_peppol_send_format_oca` based on `account_invoice_ubl_peppol` (OCA/edi#1216).

note that the reception of documents is very basic: empty draft invoices are created with the xml file and the pdf file (extracted from the xml file) attached. one could improve this by using the `account_import_ubl` module, but it is not that straightforward.

note that the tests have not been ported. they should either be removed or fixed.

huguesdk committed:
[FIX] ensure ci can run

huguesdk committed:
[MIG] backport to 12.0

* backport to 12.0.
* add new module account_peppol_send_format_oca.

huguesdk committed:
[FIX] make pre-commit pass

victor-champonnois commented on account_peppol_send_format_oca/models/account_invoice.py:
```diff
@@ -0,0 +1,88 @@
+                # ignore it)
+                if not line.product_id:
+                    raise UserError(
+                        _('Missing product on line "%s".' % line.name)
```
A more detailed error message would be nice, because this will be a surprise for the users. 
>Missing product on line "%s". Having a product on invoice lines  is necessary to correctly send the invoice through Peppol.

huguesdk replied:
good idea. done. thanks to you i also fixed all the calls to `_()`, as the formatting must be done on the translated strings, not on its source.

victor-champonnois commented on account_peppol_send_format_oca/models/account_invoice.py:
```diff
@@ -0,0 +1,88 @@
+            # issues a warning and uses payment mode 31, while for a wire
+            # transfer, payment mode 30 should be used.
+            if not self.payment_mode_id:
+                raise UserError(_("A payment mode is required."))
```
This message also happens on supplier invoices, although it should not (payment mode is not required for supplier invoices)

victor-champonnois replied:
Maybe an if condition on the type of invoice is necessary for all the checks in this methods ?

huguesdk replied:
good catch, indeed! done, thanks!

AnizR committed:
[FIX] account_peppol: CNAME to NAPTR migration
From November 1st, CNAME DNS will be deprecated for Peppol lookups. From
February 1st CNAME lookups will no longer be supported. The replacement
are NAPTR DNS records.

Multiple solutions were available, such as using DoH (e.g. with
cloudflare DNS), but we ended up choosing to proxy DNS requests through
IAP to centralize the lookups and make such specs upgrades easier to
handle in the future.

IAP is now responsible of doing the DNS lookup and fetching the service
groups of the found SMP.
cherry-pick of: https://github.com/odoo/odoo/commit/e424742c721ff05ba83f71c422594d6b5b865b49
